subsys/bluetooth/controller/ll_sw/ull_filter.c

```diff
@@ -757,10 +757,10 @@ bool ull_filter_lll_rl_addr_allowed(uint8_t id_addr_type, uint8_t *id_addr, uint
 {
 	uint8_t i, j;
 
-	/* If AR is disabled or we matched an IRK then we're all set. No hw
+	/* We matched an IRK then we're all set. No hw
 	 * filters are used in this case.
 	 */
-	if (!rl_enable || *rl_idx != FILTER_IDX_NONE) {
```

Comment:
By removing the `!rl_enable` here it means that the address comparison below can be with an entry that is in the process of being updated. When adding an RL entry ll_rl_add calls rl_access_check always allows modifications of the RL list when address resolutios is disabled, even if the adv/scan role is active.

The check below can therefore be under modification through ll_rl_add / ll_rl_remove / ll_priv_mode_set, to me this looks like it is safe, but would be nice if @carlescufi and @cvinayak could confirm this.

Reply:
rl_access_check is called with check_ar parameter set to false in the ll_rl_add/remove, which is spec requirement that the resolving list not be allowed to be modified, in this case, when the initiator is active.

Reply:
```
This command may be used at any time when address resolution is disabled in
the Controller.
```
It's different with address resolution enabled. I'm unsure if there is a test that actually checks that you can modify the RL with initiator active, and address resolution disabled, but AFAIK that should be allowed.

Reply:
If `rl_access_check` calls needs to be fixed, and if so done then the `ull_filter_lll_rl_addr_allowed` call from LLL is still safe, as the `taken` flag is checked before accessing the members of the resolving list data structure. `taken` ensures there is sane data members in the resolving list data structures. (may be need consideration of memory barrier instructions though).

In the future if a conformance test is added, or a user faces an issue, we can address the current implementation limitation of not allowing updates to resolving list, then.

Reply:
Yeah, that was my conclusion as well.